python/lsst/dax/apdb/cassandra/apdbCassandraSchema.py

```diff
@@ -360,6 +360,39 @@ def empty(self) -> bool:
                 f"Only some required APDB tables exist: {existing_tables}, missing tables: {missing_tables}"
             )
 
+    def existing_tables(self, *args: ApdbTables) -> dict[ApdbTables, list[str]]:
+        """Return the list of existing table names for given table.
+
+        Parameters
+        ----------
+        *args : `ApdbTables`
+            Tables for which to return their existing table names.
+
+        Returns
+        -------
+        tables : `dict` [`ApdbTables`, `list`[`str`]]
+            Mapping of the APDB table to the list of the existing table names.
+            More than one name can be present in the list if configuration
+            specifies per-partition tables.
+        """
+        if self._time_partition_tables and not set(args).isdisjoint(self._time_partitioned_tables):
+            # Some of the tables should have per-partition tables.
+            query = "SELECT table_name FROM system_schema.tables WHERE keyspace_name = %s"
+            result = self._session.execute(query, (self._keyspace,))
+            table_names = set(row[0] for row in result.all())
+
+            tables = {}
+            for table_enum in args:
+                if table_enum in self._time_partitioned_tables:
+                    table_pfx = table_enum.table_name(self._prefix)
+                    tables[table_enum] = [table for table in table_names if table.startswith(f"{table_pfx}_")]
+                else:
+                    tables[table_enum] = [table_enum.table_name(self._prefix)]
+            return tables
+        else:
+            # Do not check that they exist, we know that they should.
+            return {table_enum: [table_enum.table_name(self._prefix)] for table_enum in args}
```

Comment:
I think this could be factored a bit to make it easier to follow. In particular, the two calls to `table_enum.table_name(self._prefix)` in the per-partition case can be merged.

Reply:
I'm trying to figure out what you propose, can you elaborate a bit?

Reply:
The obvious factoring is:
```suggestion
            for table_enum in args:
                base_name = table_enum.table_name(self._prefix)
                if table_enum in self._time_partitioned_tables:
                    tables[table_enum] = [table for table in table_names if table.startswith(f"{base_name}_")]
                else:
                    tables[table_enum] = [base name]
            return tables
        else:
            # Do not check that they exist, we know that they should.
            return {table_enum: [table_enum.table_name(self._prefix)] for table_enum in args}
```
My gut tells me the list comprehension from the no-partition-table case can also be partially combined with the rest, but I'll admit I don't see how at the moment.

Reply:
Ah, OK, I'll add that, I thought you meant the bottom `else`.